src/main/java/org/opentripplanner/graph_builder/module/osm/FinlandWayPropertySetSource.java

```diff
@@ -24,42 +29,58 @@ public class FinlandWayPropertySetSource implements WayPropertySetSource {
+    props.setProperties("highway=living_street", withModes(ALL).bicycleSafety(0.9).walkSafety(0.9));
+    props.setProperties("highway=unclassified", withModes(ALL));
+    props.setProperties("highway=road", withModes(ALL));
+    props.setProperties("highway=byway", withModes(ALL).bicycleSafety(1.3).walkSafety(1.3));
+    props.setProperties("highway=track", withModes(ALL).bicycleSafety(1.3).walkSafety(1.3));
+    props.setProperties("highway=service", withModes(ALL).bicycleSafety(1.1).walkSafety(1.1));
+    props.setProperties("highway=residential", withModes(ALL).bicycleSafety(0.98).walkSafety(0.98));
+    props.setProperties(
+      "highway=residential_link",
+      withModes(ALL).bicycleSafety(0.98).walkSafety(0.98)
+    );
+    props.setProperties("highway=tertiary", withModes(ALL));
+    props.setProperties("highway=tertiary_link", withModes(ALL));
+    props.setProperties("highway=secondary", withModes(ALL).bicycleSafety(1.5).walkSafety(1.5));
+    props.setProperties(
+      "highway=secondary_link",
+      withModes(ALL).bicycleSafety(1.5).walkSafety(1.5)
+    );
+    props.setProperties("highway=primary", withModes(ALL).bicycleSafety(2.06).walkSafety(2.06));
+    props.setProperties(
+      "highway=primary_link",
+      withModes(ALL).bicycleSafety(2.06).walkSafety(2.06)
+    );
     // Replace existing matching properties as the logic is that the first statement registered takes precedence over later statements
-    props.setProperties("highway=trunk_link", StreetTraversalPermission.ALL, 2.06, 2.06);
-    props.setProperties("highway=trunk", StreetTraversalPermission.ALL, 7.47, 7.47);
+    props.setProperties("highway=trunk_link", withModes(ALL).bicycleSafety(2.06).walkSafety(2.06));
+    props.setProperties("highway=trunk", withModes(ALL).bicycleSafety(7.47).walkSafety(7.47));
 
     // Don't recommend walking in trunk road tunnels
-    props.setProperties("highway=trunk;tunnel=yes", StreetTraversalPermission.CAR, 7.47, 7.47);
+    props.setProperties("highway=trunk;tunnel=yes", withModes(CAR).bicycleSafety(7.47));
 
     // Do not walk on "moottoriliikennetie"
-    props.setProperties("motorroad=yes", StreetTraversalPermission.CAR, 7.47, 7.47);
+    props.setProperties("motorroad=yes", withModes(CAR).bicycleSafety(7.47));
 
     // Remove informal and private roads
-    props.setProperties("highway=*;informal=yes", StreetTraversalPermission.NONE);
-    props.setProperties("highway=service;access=private", StreetTraversalPermission.NONE);
-    props.setProperties("highway=trail", StreetTraversalPermission.NONE);
+    props.setProperties("highway=*;informal=yes", withModes(NONE));
+    props.setProperties("highway=service;access=private", withModes(NONE));
+    props.setProperties("highway=trail", withModes(NONE));
 
     // No biking on designated footways/sidewalks
-    props.setProperties("highway=footway", StreetTraversalPermission.PEDESTRIAN);
+    props.setProperties("highway=footway", withModes(PEDESTRIAN));
```

Comment:
This could have a quite low `walkSafety` value

Reply:
Indeed. I will do some adjustments to the Finland way property set source in a separate pull request.